Update Neural Network Classification Docs (#222)
ilastik/ilastik.github.io: documentation/nn/nn.md

---
# Neural Network Classification

> This workflow allows one to execute (run inference) for pre-trained convolutional neural networks (CNNs) on the data loaded into ilastik.

Reviewer: ```suggestion
This workflow allows one to execute (run inference for) pre-trained convolutional neural networks (CNNs) on the data loaded into ilastik.
```

## Prerequisites: Running locally or remotely

Unlike other ilastik workflows, the Neural Network workflow provides the possibility to run the computations on a different machine (presumably, on a remote machine with GPUs).

> If you only want to run locally, on the CPUs or GPUs of the machine were ilastik is installed, you don't need anything else, just skip to the next section. 

Reviewer: ```suggestion
If you only want to run locally, on the CPUs or GPUs of the machine where ilastik is installed, you don't need anything else, just skip to the next section. 
```

<a href="fig/load_model_01.png" data-toggle="lightbox"><img src="fig/load_model_01.png" class="img-responsive" alt="Screenshot that illustrates loading a pre-trained neural network from the Bioimage Model Zoo by specifying the doi or the nickname of the model."/></a>

Alternatively, you can download a zip archive of the model by clicking on "Download (format)".
The server will then package the model with the correct weight format for ilastik consumption.  
Once the file is downloaded you can either drag and drop the model zip file from your file explorer window into the text field, or click the arrow button while the text field is empty to bring up a load file dialog.

### 3. Run the Network

Only one thing left to do: press the "Live Predict" button to make the network predict:
<a href="fig/predicted.png" data-toggle="lightbox"><img src="fig/predicted.png" class="img-responsive" /></a>

If you prefer a different color for the display of the predictions you can right click on the corresponding layer [in the layerstack]({{site.baseurl}}/documentation/basics/layers) and select _Change color_.

### 4. Export the Result

> If you like the results, proceed to export them in the [Data Export applet]({{site.baseurl}}/documentation/basics/export). If you don't, unload this model by pressing the red cross and try out another one. 

Reviewer: ```suggestion
If you like the results, proceed to export them in the [Data Export applet]({{site.baseurl}}/documentation/basics/export). If you don't, try out another model. You can do so by first pressing the red cross below the text box to unload the model, and then the red cross inside the text box to clear your previously selected model.
```

## _Neural Network Classification (remote)_

When running remotely, the heavy lifting of neural network computations will happen on a remote machine, while you are loading the data and selecting the network on your local machine.

### On the remote machine

1. Make sure the tiktorch server and neural network framework you want to run are installed via mamba.
   Installation instructions can be found [in the tiktorch github repository](https://github.com/ilastik/tiktorch#installation).
2. Activate the tiktorch environment and start the server giving the `--addr 0.0.0.0` flag to make it accessible.

> 3. Take note of the ip/hostname of the remote machine, and the port (what out for the `Starting server on 0.0.0.0:<port>` output).

Reviewer: ```suggestion
3. Take note of the ip/hostname of the remote machine, and the port (watch out for the `Starting server on 0.0.0.0:<port>` output).
```

### On your local machine/laptop

> 

Reviewer: Why would you want to run the remote-workflow with the tiktorch server on your own machine (as a use case rather than for dev purposes)?

Author: Okay, the docs are supposed to describe what you have to do to use the remote workflow. For this you have to a) setup the server on the remote machine, and b) run the remote workflow in ilastik on your local machine. Does this make sense?
